resources/analysisTools/qcPipeline/qcJson.pl

```diff
@@ -329,20 +329,20 @@ ($$)
 
 
 sub ensureInt ($) {
-    my ($val) = @_;
-    if ($val !~ /^\d$/) {
+	my ($val) = @_;
+	if ($val !~ /^(?:\d|NaN|NA)$/) {
 		confess("Expected integer, got '$val'!");
-    }
-    return $val;
+	}
+	return $val;
 }
 
 
 sub ensureFloat ($) {
-    my ($val) = @_;
-    if ($val !~ /^(?:-?\d+(?:\.\d+)?|\d(?:\.\d+)?(?:[eE]-?\d+)?)$/) {
+	my ($val) = @_;
+	if ($val !~ /^(?:-?\d+(?:\.\d+)?|\d(?:\.\d+)?(?:[eE]-?\d+)?|NaN|NA)$/) {
```

Comment:
This checks for either a decimal followed by a point with decimals OR a decimal followed by exponent OR NaN OR NA?

Reply:
yip